test/integration/hooks/validate.test.js

```diff
@@ -41,6 +41,74 @@ describe(Support.getTestDialectTeaser('Hooks'), function() {
       });
     });
 
+    describe('#3534, hooks modifications', function() {
+      it('fields modified in hooks are saved', function() {
```

Comment:
both the issue and test deal specifically with validation hooks though. Not a major issue in any case.